devel/yarn/Portfile

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem 1.0
+
+name yarn
+version 0.17.3
+
+categories          devel
+
+platforms           darwin
+supported_archs     noarch
+
+license             BSD
+
+maintainers         andrewsforge.com:code
+
+description         JavaScript dependency manager
+long_description    Created by Facebook to replace NPM, yarn boasts about its \
+                    speed, security, and reliability. Most notably, yarn \
+                    dependencies are installed deterministically, whereas \
+                    NPM's dependencies are not.
+
+homepage            https://yarnpkg.com/
+
+master_sites        https://yarnpkg.com/downloads/${version}/
+distname            ${name}-v${version}
+worksrcdir          dist
+checksums           rmd160  be20072bfd66a6411911bebb63284abb7f7f454c \
+                    sha256  883df435f68ce47c93c2e4be27acbd4122ae52ef4d334f75104c0c5e187a9173
+
+depends_run         bin:node:nodejs6
+variant nodejs4 conflicts nodejs6 nodejs7 description {Use Node.js v4} {
+    depends_lib-replace bin:node:nodejs6 port:nodejs4
+}
```

Comment:
Likely `depends_run`, since it is node code being run. As far as versioning, isn't it the usual MacPorts pattern to include a specific one to ensure everything might run the same version of node? Like editing shebangs, configuring paths to look for libraries, etc.